src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java

```diff
@@ -97,21 +99,59 @@ public class NotificationSearchCriteria {
    */
   private String sort;
 
+  /**
+   * Retrieves the formatted from date based on the day, month, and year values.
+   *
+   * @return The formatted from date (yyyy-MM-dd), or null if the date components are not valid
+   *         integers.
+   */
+  public String getDateFrom() {
+    return getDate(notificationFromDateYear, notificationFromDateMonth, notificationFromDateDay);
+  }
+
+  /**
+   * Retrieves the formatted to date based on the day, month, and year values.
+   *
+   * @return The formatted to date (yyyy-MM-dd), or null if the date components are not valid
+   *         integers.
+   */
+  public String getDateTo() {
+    return getDate(notificationToDateYear, notificationToDateMonth, notificationToDateDay);
+  }
+
+  private String getDate(String yearInput, String monthInput, String dayInput) {
```

Comment:
might be worth just adding a javadoc to this private method

Reply:
Added